src/components/form/InputNumberField.tsx

```diff
@@ -1,28 +1,91 @@
+import { useState } from "react";
 import { useTranslation } from "react-i18next";
 import { useField } from "formik";
 import FormItem from "antd/es/form/FormItem";
-import { InputNumber, InputNumberProps, message } from "antd";
-import { ChangeEvent } from "react";
-import { defaultAPIConfig } from "../../config/defaultApiConfig";
+import { InputNumber, InputNumberProps, message, Checkbox } from "antd";
 import { TeddyCloudApi } from "../../api";
+import { defaultAPIConfig } from "../../config/defaultApiConfig";
 
 type InputNumberFieldProps = {
   name: string;
   label?: string;
   description?: string;
+  overlayed?: boolean;
+  overlayId?: string;
 };
 
 const InputNumberField = (props: InputNumberFieldProps & InputNumberProps) => {
   const { t } = useTranslation();
-  const { name, label, description, ...inputNumberProps } = props;
+  const { name, label, description, overlayed: initialOverlayed, overlayId, ...inputNumberProps } = props;
   const [field, meta, helpers] = useField<number | undefined>(name!);
+  const [overlayed, setOverlayed] = useState(initialOverlayed); // State to track overlayed boolean
 
   const hasFeedback = !!(meta.touched && meta.error);
   const help = meta.touched && meta.error && t(meta.error);
   const validateStatus = meta.touched && meta.error ? "error" : undefined;
+  const addonAfter = overlayed === undefined ? "" : (
+    <Checkbox
+      checked={overlayed}
+      onChange={(e) => {
+        setOverlayed(e.target.checked);
+        handleOverlayChange(e.target.checked);
+      }}
+    >
+      {t("settings.overlayed")}
+    </Checkbox>
+  );
 
   const api = new TeddyCloudApi(defaultAPIConfig());
 
+  const handleOverlayChange = (checked: boolean) => {
+    const overlayRoute = `?overlay=${overlayId}`;
+    const url = `${process.env.REACT_APP_TEDDYCLOUD_API_URL}/api/settings/${checked ? "set" : "reset"}/${name}${overlayRoute}`;
```

Comment:
this code is used on multiple places in the same way. therefor it could be written in one place and be referenced

Reply:
i am with you, good task for a refactoring. but not here now